packages/react-dom/src/client/ReactInputSelection.js

```diff
@@ -49,7 +49,25 @@ function containsNode(outerNode, innerNode) {
 }
 
 function isInDocument(node) {
-  return containsNode(document.documentElement, node);
+  return (
+    node &&
+    node.ownerDocument &&
+    containsNode(node.ownerDocument.documentElement, node)
+  );
+}
+
+function getActiveElementDeep() {
+  let win = window;
+  let element = getActiveElement();
+  while (element instanceof win.HTMLIFrameElement) {
+    try {
+      win = element.contentDocument.defaultView;
+    } catch (e) {
```

Comment:
What exactly are we catching here? Is it `defaultView` access throwing? Or do we expect `contentDocument` to potentially be missing?

Reply:
If you just try to access the contentDocument of an `HTMLIframeElement` that comes from a non-matching (separate) domain (like if there’s an embedded YouTube player in the DOM), the browser will throw an error.

Reply:
Could you add a comment to this line describing such a scenario?